Assignment 2.Rmd

```diff
@@ -87,11 +92,54 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
-
+```{r}
+student_sum<- EDGE %>% group_by(to) %>% summarise(sum(count))
+names(student_sum)<-c("id","count")
+teacher_sum<-EDGE %>%group_by(from)%>%summarise(sum(count))
+names(teacher_sum)<-c("id","count")
+vertex_new<-bind_rows(student_sum,teacher_sum)
+vertex_new
+
+vertex1
+g1 <- graph.data.frame(EDGE, directed=TRUE, vertices=vertex_new)
+plot(g1,layout=layout.fruchterman.reingold, 
+     edge.width=EDGE$count,
+     vertex.size = vertex_new$count*1.5)
+```
+```{r}
+plot(g,layout=layout.fruchterman.reingold, vertex.size=EDGE$count*3,
+     vertex.color=VERTEX$gender,
+     edge.width=EDGE$count)
+```
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+data<-select()
+```{r}
+hudk4050.classes
+
+data<-select(hudk4050.classes,V20,V21,V22,V23,V24,V25,V26)
+data<-data[-c(1,2,3),]
+Name<-data[,1]
+data[,1]<-c(1:53)
+colnames(data)<-c("Name","Class 1","Class 2","Class 3","Class 4","Class 5","Class 6")
+
+datatran<-gather(data,"Class Number","ClassName",2:7)
+datatran<-datatran[,-2]
+datatran<-filter(datatran,ClassName> 0)#,ClassName!="HUDK4050",ClassName!="HUDK 4050",ClassName!="4050")
+datatran$count<-1
+dataaa<-spread(datatran,ClassName,count,0)
+row.names(dataaa)<-Name
+dataaa<-dataaa[,-1]
+datafinal<-as.matrix(dataaa)
+adj<-datafinal %*%t(datafinal)
+diag(adj)<-0
+g <- graph.adjacency(adj,mode="undirected")
+plot(g,layout=layout.fruchterman.reingold, vertex.size=2)
+
+```
+
```

Comment:
Can you explain the difference between the two types of centrality measure? Which one would you prefer in this case as a measure of the most central student?